src/generic/stage2/numberparsing.h

```diff
@@ -90,14 +90,24 @@ simdjson_really_inline bool compute_float_64(int64_t power, uint64_t i, bool neg
 
   // We are going to need to do some 64-bit arithmetic to get a more precise product.
   // We use a table lookup approach.
-  components c =
-      power_of_ten_components[power - FASTFLOAT_SMALLEST_POWER];
-      // safe because
-      // power >= FASTFLOAT_SMALLEST_POWER
-      // and power <= FASTFLOAT_LARGEST_POWER
-  // we recover the mantissa of the power, it has a leading 1. It is always
+  // It is safe because
+  // power >= FASTFLOAT_SMALLEST_POWER
+  // and power <= FASTFLOAT_LARGEST_POWER
+  // We recover the mantissa of the power, it has a leading 1. It is always
   // rounded down.
-  uint64_t factor_mantissa = c.mantissa;
+  uint64_t factor_mantissa = mantissa_64[power - FASTFLOAT_SMALLEST_POWER];
+  
+  // For q in (-400,350), we have that
+  // f = (((152170 + 65536) * q ) >> 16);
+  // is equal to
+  //  floor(p) + q
+  // where
+  //   p = log(5**q)/log(2) = q * log(5)/log(2)
+  //
+  // Note that this is not magic: 152170/(1<<16) is approximatively equal to log(5)/log(2).
+  //
+  int64_t exponent = (((152170 + 65536) * power) >> 16) + 1024 + 63;
```

Comment:
I can't say this makes it the algorithm *clearer*, but it definitely makes it less magic!

If you want to make things clearer, here are the bits that aren't obvious to me (maybe they will be to all float people):
* One-character names: good for math, not as much for code :)
  - `q` took me some time to figure out, but given it's the same as `power`, should we use that name in the comment?
  - `f` I still don't know, except that I can see that it's a `exponent` shifted by 1024+63. I don't know why though :)
  - `p`: is that the same as the `p` in the earlier comment? It looks like p^10 * 2 = q^10 * 5, so I can deduce that p and q are both power-type numbers ... but that's as far as I got looking at this.
* 152170/(1<<16) being equal to log(5)/log(2) is super useful, but I'm not sure where the 65536 came from :)
* Not sure why the 1024 and 63 are there?

Anyway, this is definitely better than before, so take my layman's comments with a grain of salt. I can only ask these questions now because you made it less opaque :)

Reply:
I'll elaborate and merge. The issue of providing better documentation is on my mind and it will be done (beyond these comments.)

Reply:
Ok. I added generous comments. To properly explain the whole thing, more work is needed.

Reply:
Thanks! :)